@here/harp-lines/lib/Lines.ts

```diff
@@ -41,20 +41,21 @@ export function createLineGeometry(
     let sum = 0;
     L[0] = 0;
     for (let i = 0; i < pointCount - 1; ++i) {
-        const dx = polyline[(i + 1) * 2] - polyline[i * 2];
-        const dy = polyline[(i + 1) * 2 + 1] - polyline[i * 2 + 1];
+        const dx = polyline[(i + 1) * 3] - polyline[i * 3];
+        const dy = polyline[(i + 1) * 3 + 1] - polyline[i * 3 + 1];
         const len = Math.sqrt(dx * dx + dy * dy);
         sum = sum + len;
         L[i + 1] = sum;
         tangents[i] = [dx, dy];
     }
 
     const isClosed =
-        polyline[0] === polyline[polyline.length - 2] &&
-        polyline[1] === polyline[polyline.length - 1];
+        polyline[0] === polyline[polyline.length - 3] &&
```

Comment:
An alternative approach that could save same memory for lines projected on 2D flat surfaces (e.g. mercator projected without terrain) would be to pass the dimensions (2 or 3) to createLineGeometry, in a way making it similar to the eartcut triangulation APIs. Of course, this will require creating two different vertex formats for 2D and 3D lines also the line shader should define the `attribute position` using vec2 or vec3 depending on the dimension.

Reply:
Again, not a bad point, but more in the scope of ``harp-lines`` refactoring ticket...

Reply:
I agree, we should remove @here/harp-lines and introduce a more generic @here/harp-geometry with all the bits needed to work with geometries, and yes, I am with you, this is definitely not something that we should do in this PR. 

That said, this change is touching createLineGeometry(...) and it is changing its signatures from accepting a sequence of packaged 2D coordinates (something that this function knows how to handle) to a sequence of 3D points. Unfortunately, this new version of createLineGeometry, at least for now, doesn't create correct geometries unless all the input 3D points have the same Z component, and I find this wrong. In my opinion we should change the signature of createLineGeometry to accept 3D points only after we verified that its code can create valid geometries.